src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -861,8 +861,9 @@ DeferredErrorIfUnsupportedRecurringTuplesJoin(
 	{
 		return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
 							 "cannot pushdown the subquery",
-							 "Complex subqueries and CTEs cannot be in the outer "
-							 "part of the outer join", NULL);
+							 "Complex subqueries, CTEs and non-distributed tables"
+							 "(only if the inner part includes a distributed table) "
```

Comment:
The part in parentheses seems a bit hard to digest

Maybe we can change the line below to "cannot be in the outer part of an outer join with a distributed table"